slidechain/cmd/peg/peg.go

```diff
@@ -99,16 +95,18 @@ func main() {
 		}
 	}
 
-	assetXDR, err := asset.MarshalBinary()
+	amountXLM, err := xlm.Parse(*amount)
```

Comment:
I believe this only applies in the case of lumens, meaning this belongs in the `else` clause just above. For other asset types you need to do different parsing. I think it's sufficient in those cases to do `strconv.ParseInt`.

Reply:
Discussed offline - for posterity, from https://www.stellar.org/developers/guides/concepts/assets.html#amount-precision-and-representation it appears that all assets are scaled down by a factor of ten million.

Reply:
So 10000000 USD is one dollar. What could go wrong? 🙄